CORE-5324 Correct Cpi/ Cpb v1 format in `packaging-verify` library

kyriathar opened the pull request:
Currently `CpbV1Verifier` expects a Cpb v1 to contain a group policy file. Even though in v1 we amend the Cpb file by adding the group policy file to become the Cpi, the check for group policy file should not be happening at  `CpbV1Verifier` level but rather at `CpiV1Verifier` level.

- moves group policy file check from `CpbV1Verifier` to `CpiV1Verifier`.
- removes from `TestCpbV1Builder` adding of group policy file.
- adds `TestCpiV1Builder` which creates a Cpb and then it amends it to include group policy file.

mbrkic-r3 commented on libs/packaging/packaging-verify/src/test/kotlin/net/corda/libs/packaging/verify/internal/cpi/CpiV1VerifierTest.kt:
```diff
@@ -31,9 +32,14 @@ class CpiV1VerifierTest {
+            TestCpiV1Builder()
+                .cpb(
+                    TestCpbV1Builder()
+                        .signers(ALICE)
```
Something like this could be added to `TestCpiV1Builder` so we don't need to specify signers for every layer if they are same:
https://github.com/corda/corda-runtime-os/blob/cb171dce5fc0aac59ce650cd5fc9e1c2b4ac8896/testing/packaging-test-utilities/src/main/kotlin/net/corda/libs/packaging/testutils/cpi/TestCpiV2Builder.kt#L38

kyriathar replied:
Yeah good point. This is now added.

mbrkic-r3 commented on testing/packaging-test-utilities/src/main/kotlin/net/corda/libs/packaging/testutils/cpi/TestCpiV1Builder.kt:
```diff
@@ -0,0 +1,27 @@
+
+    fun build() =
+        cpb.build().apply {
+            addFile(TestCpiV2Builder.POLICY_FILE, policy)
```
`TestCpiV2Builder. `can be removed

kyriathar replied:
Good catch, removed

kyriathar-r3 committed:
CORE-5324 Correct Cpi/ Cpb v1 format in `packaging-verify` library (#1821)

- moves group policy file check from `CpbV1Verifier` to `CpiV1Verifier`.
- removes from `TestCpbV1Builder` adding of group policy file.
- adds `TestCpiV1Builder` which creates a Cpb and then it amends it to include group policy file.
